### PR TITLE
RFC: Move Peer and PeerManagerImpl declarations to separate header

### PR DESCRIPTION
**Pull request by jnewbery**
Not sure whether this is desirable, so marking as a proof of concept.

This moves the `Peer` and `PeerManagerImpl` declarations to their own header file, `peerman_impl.h`, which can be included by `net_processing.cpp` and the test/bench/fuzz files.

The benefits of this are:

- `PeerManagerImpl` functions which are exposed through the `PeerManager` interface for testing, but would otherwise be private can be removed from `PeerManager`. That means that `PeerManager` truly is net_processing's minimal interface to expose to the rest of the program.
- If a test needs to directly manipulate `Peer` objects, it can do so, since they're exposed in peerman_impl header.

### CODE REVIEW

**Comment by jnewbery** [comment]
This is a minimal implementation of what was first discussed here: https://github.com/bitcoin/bitcoin/pull/20758#issuecomment-751358315. Thoughts, @ajtowns?

File: src/Makefile.am [hunk 1]
```diff
@@ -179,6 +179,7 @@ BITCOIN_CORE_H = \
   noui.h \
   optional.h \
   outputtype.h \
+  peerman_imlp.h \
```
Comment (ajtowns) [inline src/Makefile.am:182]: "impl"

**Review by ajtowns — COMMENTED** [review]
I'm not going to think seriously about this until having tried to pull orphan handling or similar into its own module, in the hopes that that alone will be enough to make testing work better.

File: src/Makefile.am (hunk 1 above)
Comment (jnewbery) [reply src/Makefile.am:182]: gah

**Comment by jnewbery** [comment]
Fixed the typo. I'll fix the linter and fuzzer errors if this PR gets some concept ACKs.

**Comment by jnewbery** [comment]
Closing this for now. Let's revisit once #21148 is merged.

**Pull request closed without merging**